normandy/recipes/migrations/0035_revision_data.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.9.11 on 2016-11-28 13:48
+# flake8: noqa
+from __future__ import unicode_literals
+
+import hashlib
+import json
+
+from django.db import migrations
+
+
+def hash_revision_data(revision):
+    data = '{}{}{}{}{}{}'.format(revision.recipe.id, revision.created, revision.name,
+                                 revision.action.id, revision.arguments_json,
+                                 revision.filter_expression)
+    return hashlib.sha256(data.encode()).hexdigest()
+
+
+def migrate_reversion_data(apps, schema_editor):
+    Action = apps.get_model('recipes', 'Action')
+    Recipe = apps.get_model('recipes', 'Recipe')
+    RecipeRevision = apps.get_model('recipes', 'RecipeRevision')
+    Version = apps.get_model('reversion', 'Version')
+    ContentType = apps.get_model('contenttypes', 'ContentType')
+
+    try:
+        ct = ContentType.objects.get(app_label='recipes', model='recipe')
+    except ContentType.DoesNotExist:
+        # There are no content types so there is no data to be migrated
+        pass
+    else:
+        for r in Recipe.objects.all():
+            revision = None
+
+            for v in Version.objects.filter(content_type=ct, object_id=r.id).order_by(
+                    'revision__date_created'):
+
+                v = Version.objects.get(revision_id=v.revision.id)
+
+                data = json.loads(v.serialized_data)[0]['fields']
+
+                action = Action.objects.get(id=data.get('action'))
+
+                revision = RecipeRevision(
+                    recipe=r, parent=revision, created=v.revision.date_created,
+                    comment=v.revision.comment, user=v.revision.user, name=data.get('name'),
+                    action=action, arguments_json=data.get('arguments_json', ''),
+                    filter_expression=data.get('filter_expression', ''))
+
+                revision.id = hash_revision_data(revision)
+
+                revision.save()
+
+            r.latest_revision = revision
```

Comment:
It should be creating a reversion when the recipe is created. So that's not it. I'll add the `assert`.